### PR TITLE
Added functionality in show_gpus to have an option to show all regions' pricing for an accelerator

### PR DESCRIPTION
**Pull request by Vaibhav2001**
This solves #2863 

show_gpus can now accept a new flag `--all-regions`, which displays all the regions for a given accelerator. It can be used as follows. 

`sky show-gpus <acclerator> --all-regions`

Additionally it can take a cloud argument as well. 

`sky show-gpus <acclerator> --cloud <cloud> --all-regions`

I compared the outputs against the catalog. 

Tested (run the relevant ones):

- [x] Code formatting: `bash format.sh`
- [x] Checking all regions exist.

### CODE REVIEW

File: sky/cli.py [hunk 1]
```diff
@@ -3390,13 +3391,18 @@ def check(verbose: bool):
     ('The region to use. If not specified, shows accelerators from all regions.'
     ),
 )
+@click.option('--all-regions',
+              is_flag=True,
+              default=False,
+              help='Shows details for all regions.')
```
Comment (romilbhardwaj) [inline sky/cli.py:3397]:
```suggestion
              help='Show pricing and instance details for a specified accelerator across all regions and clouds.')
```

File: sky/cli.py [hunk 2]
```diff
@@ -3410,6 +3416,9 @@ def show_gpus(
     To show all accelerators, including less common ones and their detailed
     information, use ``sky show-gpus --all``.
 
+    To show all regions for a specified accelerator, use
+    ``sky show-gpus <accelerator> --all-regions``.
+
```
Comment (romilbhardwaj) [inline sky/cli.py:3421]: Can we also update the docstr at L3429:

```
If ``--region`` or ``--all-regions`` is not specified, ...
```

**Review by romilbhardwaj — COMMENTED** [review]
Thanks @Vaibhav2001! Seems to work nicely. Left some comments.

File: sky/cli.py [hunk 3]
```diff
@@ -3425,6 +3434,11 @@ def show_gpus(
     if region is not None and cloud is None:
         raise click.UsageError(
             'The --region flag is only valid when the --cloud flag is set.')
+
+    # validation for the --region flag
+    if all_regions and accelerator_str is None:
+        raise click.UsageError('The --all-region flag is only valid when an \
+            accelerator is specified.')
```
[the remark this answers is not in the document]
Comment (Vaibhav2001) [reply sky/cli.py:3441]: Makes sense! Will fix

File: sky/clouds/service_catalog/__init__.py [hunk 4]
```diff
@@ -47,7 +46,11 @@ def _map_clouds_catalog(clouds: CloudFilter, method_name: str, *args, **kwargs):
             raise AttributeError(
                 f'Module "{cloud}_catalog" does not '
                 f'implement the "{method_name}" method') from None
-        results.append(method(*args, **kwargs))
+        if cloud == 'kubernetes' and method_name == 'list_accelerators':
+            #remove the last argument (all_regions) for kubernetes
+            results.append(method(*args[:-1], **kwargs))
+        else:
+            results.append(method(*args, **kwargs))
```
[the remark this answers is not in the document]
Comment (Vaibhav2001) [reply sky/clouds/service_catalog/__init__.py:53]: That is what I was trying to do. However pylint throws an error for an unused variable. So I left it like this instead

File: sky/cli.py [hunk 5]
```diff
@@ -795,7 +795,8 @@ def _launch_with_confirm(
 
     maybe_status, _ = backend_utils.refresh_cluster_status_handle(cluster)
     if maybe_status is None:
-        # Show the optimize log before the prompt if the cluster does not exist.
+        # Show the optimize log before the prompt if the cluster does not
+        # exist.
```
[the remark this answers is not in the document]
Comment (Vaibhav2001) [reply sky/cli.py:799]: Do you mean the reformatting onto two lines?

File: sky/cli.py [hunk 6]
```diff
@@ -3425,6 +3434,11 @@ def show_gpus(
     if region is not None and cloud is None:
         raise click.UsageError(
             'The --region flag is only valid when the --cloud flag is set.')
+
+    # validation for the --region flag
```
[the remark this answers is not in the document]
Comment (Vaibhav2001) [reply sky/cli.py:3438]: Agreed

File: sky/cli.py [hunk 7]
```diff
@@ -4279,7 +4294,8 @@ def serve_up(
                     'will use the port specified as application ingress port.')
         service_port_str = requested_resources.ports[0]
         if not service_port_str.isdigit():
-            # For the case when the user specified a port range like 10000-10010
+            # For the case when the user specified a port range like
+            # 10000-10010
```
[the remark this answers is not in the document]
Comment (Vaibhav2001) [reply sky/cli.py:4298]: Yeah for some reason `yapf` did this

File: sky/clouds/service_catalog/__init__.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (Vaibhav2001) [reply sky/clouds/service_catalog/__init__.py:53]: @romilbhardwaj lmk if this is fine or if there is a way to suppress the error.

[the remark this answers is not in the document]
Comment (romilbhardwaj) [reply sky/clouds/service_catalog/__init__.py:53]: Yes, I think `del <unused variable` should be fine as we do in rest of our code. If you've resolved, can you push your changes? I think the branch is not upto date..

[the remark this answers is not in the document]
Comment (Vaibhav2001) [reply sky/clouds/service_catalog/__init__.py]: Got it! Yeah sorry I was waiting to push after addressing all comments.

**Review by romilbhardwaj — APPROVED** [review]
Thanks @Vaibhav2001! Tested with clouds + a Kubernetes T4 cluster, seems to be working well. LGTM.

**Comment by concretevitamin** [comment]
Awesome work @Vaibhav2001.

For the same `gpu:count`, it may be good to sort by `(HOURLY_PRICE, HOURLY_SPOT_PRICE, REGION)`. The current behavior appears to be sorting by REGION.
```
» sky show-gpus --all-regions A100 --cloud gcp
*NOTE*: for most GCP accelerators, INSTANCE_TYPE == (attachable) means the host VM's cost is not included.

GPU   QTY  CLOUD  INSTANCE_TYPE   DEVICE_MEM  vCPUs  HOST_MEM  HOURLY_PRICE  HOURLY_SPOT_PRICE  REGION
A100  1    GCP    a2-highgpu-1g   -           12     85GB      $ 3.841       $ 1.536            asia-northeast1
A100  1    GCP    a2-highgpu-1g   -           12     85GB      $ 3.841       $ 1.536            asia-northeast3
A100  1    GCP    a2-highgpu-1g   -           12     85GB      $ 3.841       $ 1.536            asia-southeast1
A100  1    GCP    a2-highgpu-1g   -           12     85GB      $ 3.673       $ 1.176            europe-west4
A100  1    GCP    a2-highgpu-1g   -           12     85GB      $ 3.967       $ 1.587            me-west1
A100  1    GCP    a2-highgpu-1g   -           12     85GB      $ 3.673       $ 1.469            us-central1
A100  1    GCP    a2-highgpu-1g   -           12     85GB      $ 3.673       $ 1.469            us-east1
A100  1    GCP    a2-highgpu-1g   -           12     85GB      $ 3.673       $ 1.469            us-west1
A100  1    GCP    a2-highgpu-1g   -           12     85GB      $ 4.176       $ 1.670            us-west3
A100  1    GCP    a2-highgpu-1g   -           12     85GB      $ 3.841       $ 1.536            us-west4
```

**Pull request merged**